app/controller/List.js

```diff
@@ -337,24 +357,35 @@ Ext.define('WhatsFresh.controller.List', {
             loc: "the database",
             end: "."
         };
-        
-        // Location/City is specified:
-        // "There are <number> vendors near <location>."
-        if (locationString !== "Please choose a location"){
-            summary.i = "near ";
-            summary.loc = locationString;
-        }
+
+        // User Location specified:
+        // "There are <number> vendors near you."
+        if(userLoc === 1){
+        	summary.i = "near ";
+        	summary.loc = "you";
+        }else{
+	        // Location/City is specified:
+	        // "There are <number> vendors near <location>."
+	        if (locationString !== "Please choose a location"){
+	            summary.i = "near ";
+	            summary.loc = locationString;
+	        }
+	    }
 
         // Product is specified:
         // "There are <number> vendors ... with <product>."
         if (productString !== "Please choose a product"){
+        	if((userLoc !== 1) & (locationString === "Please choose a location")){
+	        	summary. i = null;
+	        	summary.loc = null;
+	        }
```

Comment:
This reads poorly. We should not have to undo our work from the previous step, lines 361-373.
The case where the user has not selected either their device or a city to filter by does not depend on whether or not they have also selected a product.

I recommend including this logic in the previous statement OR resetting the defaults at the beginning of this method.